pkg/network/devices.go

```diff
@@ -0,0 +1,51 @@
+package network
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+)
+
+// GetFreeDeviceName returns a device name that is unused; used when no network
+// name is provided
+func GetFreeDeviceName() (string, error) {
+	var (
+		deviceNum  uint
+		deviceName string
+	)
+	networkNames, err := GetNetworkNamesFromFileSystem()
+	if err != nil {
+		return "", err
+	}
+	liveNetworksNames, err := GetLiveNetworkNames()
+	if err != nil {
+		return "", err
+	}
+	for {
+		deviceName = fmt.Sprintf("%s-%d", CNIDeviceName, deviceNum)
+		logrus.Debugf("checking if device name %s exists in other cni networks", deviceName)
+		if StringInSliceContains(deviceName, networkNames) {
+			deviceNum++
+			continue
+		}
+		logrus.Debugf("checking if device name %s exists in live networks", deviceName)
+		if !StringInSliceContains(deviceName, liveNetworksNames) {
+			break
+		}
+		// TODO Still need to check the bridge names for a conflict but I dont know
+		// how to get them yet!
+		deviceNum++
+	}
+	return deviceName, nil
+}
+
+// StringInSliceContains returns a bool if the given string is found
+// in the given string slice
+func StringInSliceContains(term string, group []string) bool {
```

Comment:
Let's use `pkg/util/StringInSlice()` instead. I think such a generic helper doesn't belong in a specialised package.